### ISSUE TITLE
npm3 dependency resolution: shared dependencies

### ISSUE THREAD

**Issue by droidenator** [npm/docs#516]
I'm looking at the docs found here: https://docs.npmjs.com/how-npm-works/npm3

If I was to install module D that had a dependency on module B v1.0, would it use the top level module B since the versions match?

And kind of an extension on this question: let's say I install module E that requires module A (which I've installed myself); will npm still use module A for the dependency if the versions match? Would there be any issues if I upgraded to a new version of module A? Would npm know to install the required version of module A under module E to prevent issues?

**Comment by slorber** [comment]
Hi,

I don't know if I should open another issue for that but on the same doc page:

![dep](https://docs.npmjs.com/images/npm3deps4.png)


If A depends on B1 and C depends on B2, why would B1 be a first level dependency while B2 is a 2nd level dependency?

**Comment by ashleygwilliams** [comment]
hi @droidenator and @slorber i just shipped a new doc page here => https://docs.npmjs.com/how-npm-works/npm3-dupe

could you let me know if that helps or if you have further questions? thanks!

**Comment by ashleygwilliams** [comment]
clsoing. please reopen if you still have questions!

**Comment by slorber** [comment]
I did take a glance and it seems better explained (but it does not seem to contain the initial case I put in my former comment)

**Comment by droidenator** [comment]
@ashleygwilliams it looks like the new documentation covers everything I had questions about. Thanks!

@slorber I think I can answer your question and (I think) it is addressed in the docs but maybe not as directly as you were looking for. In short, install order matters. `package.json` installs modules in alphabetical order. 

Using the diagram you included in your comment, the first module installed with a dependency on B will determine which version of B is installed as a first level dependency. Since A depends on B1, B1 becomes the first level dependency. If module A was renamed to module D, then module C will be the first module installed and the first level dependency would be B2. Module D would end up with B1 installed as a second level dependency.
